### PR TITLE
Hud: Added perks to the UI

### PR DESCRIPTION
**Pull request by Alf21**
image

### CODE REVIEW

File: code/items/perks/Radar1.cs [hunk 1]
```diff
@@ -0,0 +1,15 @@
+using Sandbox;
+
+namespace TTTReborn.Items
+{
+    [Library("ttt_radar1")]
+    public class Radar1 : TTTPerk, IBuyableItem
```
Comment (mzegar) [inline code/items/perks/Radar1.cs:6]: What's the difference between `Radar1` vs. `Radar2`?

File: code/ui/Hud.cs [hunk 2]
```diff
@@ -34,10 +33,14 @@ public static void OnHotReloaded()
...
             {
                 Local.Hud?.Delete();
+
                 Hud hud = new Hud();
+
                 if (Local.Client.Pawn is TTTPlayer player && player.LifeState == LifeState.Alive)
                 {
                     hud.AliveHudPanel.CreateHud();
+
+                    hud.AliveHudPanel.Effects.OnHotReloaded();
```
Comment (mzegar) [inline code/ui/Hud.cs:43]: Can't we create the elements inside of `CreateHud()`?

File: code/ui/Hud.cs [hunk 3]
```diff
@@ -95,13 +100,16 @@ public void CreateHud()
             {
                 if (_panels.Count == 0)
                 {
+                    Effects = Parent.AddChild<Effects>();
```
Comment (mzegar) [inline code/ui/Hud.cs:103]: If we remove the earlier `Effects.OnHotReloaded` then we don't have to uniquely handle `Effects` here right?

**Review by mzegar — CHANGES_REQUESTED** [review]
If you need to work off a branch, you can just make a branch of a branch. This would be helpful since I had to skim through a lot of this stuff in a different PR. 

https://stackoverflow.com/questions/4470523/create-a-branch-in-git-from-another-branch

42 files is a lot : D

File: code/items/perks/Radar1.cs (hunk 1 above)
Comment (Alf21) [reply code/items/perks/Radar1.cs:6]: Just testing the UI stuff

File: code/ui/Hud.cs (hunk 2 above)
Comment (Alf21) [reply code/ui/Hud.cs:43]: We don't know whether it's hotloaded then :o

File: code/ui/Hud.cs (hunk 3 above)
Comment (Alf21) [reply code/ui/Hud.cs:103]: We need this because it's e.g. used in `TTTPerks.OnEquip`

**Review by mzegar — APPROVED** [review]
Made some changes, check em over.

File: code/ui/Effects.cs [hunk 4]
```diff
@@ -8,45 +8,40 @@
...
+        public void AddEffect(TTTPerk perk)
         {
-            PerksInventory perks = ((Local.Pawn as TTTPlayer).Inventory as Inventory).Perks;
+            _effectList.Add(new Effect(this) {item = perk});
+        }
 
-            for (int i = 0; i < perks.Count(); i++)
+        public void RemoveEffect(TTTPerk perk)
+        {
+            foreach (Effect effect in _effectList.Where(effect => effect.item.Name == perk.Name))
```
Comment (Alf21) [inline code/ui/Effects.cs]: Just one item of the same name can exist in a list, so we don't need a foreach loop theoretically. Maybe merge everything using Linq? :)

**Pull request merged**